akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,16 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case _ ⇒
+        val conn = url.openConnection()
+        try {
+          conn.setUseCaches(false) // otherwise the JDK will keep the JAR file open when we close!
+          val len = conn.getContentLength
+          val lm = conn.getLastModified
+          Some(ResourceFile(url, len, lm))
+        } catch {
+          case _: Throwable ⇒ None
```

Comment:
Dunno if we should swallow everything here.

To defend against a possible `NullPointerException` and detect directories (`getInputStream == null`) you could use:
```scala
for {
  c <- Option(url.openConnection())
  _ = c.setUseCaches(false) // otherwise the JDK will keep the resource open when we close!
  is <- Option(c.getInputStream)
} yield {
  is.close()
  ResourceFile(url, c.getContentLength, c.getLastModified)
}
```